sys/net/gnrc/sock/ip/gnrc_sock_ip.c

```diff
@@ -129,7 +130,13 @@ ssize_t sock_ip_recv_buf_aux(sock_ip_t *sock, void **data, void **buf_ctx,
         return -EADDRNOTAVAIL;
     }
     tmp.family = sock->local.family;
-    res = gnrc_sock_recv((gnrc_sock_reg_t *)sock, &pkt, timeout, &tmp);
+#if IS_USED(MODULE_SOCK_AUX_LOCAL)
+    if ((aux != NULL) && (aux->flags & SOCK_AUX_GET_LOCAL)) {
+        _aux.local = &aux->local;
+        aux->flags &= ~(SOCK_AUX_GET_LOCAL);
```

Comment:
Shouldn't this flag be unset _after_ `gnrc_sock_recv()` returned a non-error?

Reply:
This is a bit more efficient like this. I extended the API doc to specify that the contents of `aux` are invalid when the function call returns with an error code.

Reply:
I agree with your workaround for the `port` member in UDP. But I as a user would be a bit surprised (and annoyed) if a flag, I previously set, would be now unset (especially if not mentioned explicitly in the doc, but even then), even if the doc said the content is undefined.

Reply:
OK. But then I can also populate port as well only on success, as the cost for a second if was already paid.